### ISSUE TITLE
httpserver.cpp:58:25: error: ‘MHD_USE_EPOLL_INTERNALLY’ was not declared in this scope

### ISSUE THREAD

**Issue by drandreaskrueger** [cinemast/libjson-rpc-cpp#180]
```
git clone git://github.com/cinemast/libjson-rpc-cpp.git
mkdir -p libjson-rpc-cpp/build
cd libjson-rpc-cpp/build
cmake ..
```

results in 

```

[ 24%] Building CXX object src/jsonrpccpp/CMakeFiles/jsonrpcserver.dir/server/rpcprotocolserverv1.cpp.o
[ 25%] Building CXX object src/jsonrpccpp/CMakeFiles/jsonrpcserver.dir/server/connectors/httpserver.cpp.o
/home/akrueger/Documents/Code/eth/libjson-rpc-cpp/src/jsonrpccpp/server/connectors/httpserver.cpp: In member function ‘virtual bool jsonrpc::HttpServer::StartListening()’:
/home/akrueger/Documents/Code/eth/libjson-rpc-cpp/src/jsonrpccpp/server/connectors/httpserver.cpp:58:25: error: ‘MHD_USE_EPOLL_INTERNALLY’ was not declared in this scope
             mhd_flags = MHD_USE_EPOLL_INTERNALLY;
                         ^
src/jsonrpccpp/CMakeFiles/jsonrpcserver.dir/build.make:206: recipe for target 'src/jsonrpccpp/CMakeFiles/jsonrpcserver.dir/server/connectors/httpserver.cpp.o' failed
make[2]: *** [src/jsonrpccpp/CMakeFiles/jsonrpcserver.dir/server/connectors/httpserver.cpp.o] Error 1
CMakeFiles/Makefile2:176: recipe for target 'src/jsonrpccpp/CMakeFiles/jsonrpcserver.dir/all' failed
make[1]: *** [src/jsonrpccpp/CMakeFiles/jsonrpcserver.dir/all] Error 2
Makefile:160: recipe for target 'all' failed
make: *** [all] Error 2
```

on Debian 8.6

**Comment by cinemast** [comment]
Thanks for reporting this. Debian has a rather old version of libmicrohttpd which das not support EPOLL.
The above mentioned commit should fix this.

**Comment by martijnmumbles** [comment]
I pulled libmicrohttpd-latest.tar.gz from http://ftp.gnu.org/gnu/libmicrohttpd/ today and had the exact same issue.
MHD_USE_EPOLL_INTERNALLY_LINUX is defined, but MHD_USE_EPOLL_INTERNALLY isn't.

Same fix bypasses it of course, just figured I'd add on that it doesn't seem to be limited to old versions of libmicrohttpd.

**Comment by metalicjames** [comment]
I also have this problem on Ubuntu 16.04

**Comment by cinemast** [comment]
Did you try the devlop branch containing this fix?

**Comment by porunov** [comment]
I have the same problem on Debian 8

**Comment by cinemast** [comment]
I will prepare a fresh release within the next days. It will build on various paltforms during CI process: https://travis-ci.org/cinemast/libjson-rpc-cpp/builds/220214611

**Comment by kamilosxd678** [comment]
I found something like this
libmicrohttpd-0.9.53/src/include/microhttpd.h:1041:  _MHD_DEPR_IN_MACRO("Value MHD_USE_EPOLL_INTERNALLY is deprecated, use MHD_USE_EPOLL_INTERNAL_THREAD")

And using MHD_USE_EPOLL_INTERNAL_THREAD helps

**Comment by chfast** [comment]
The fix for Ubuntu 16.04 is in #195.

**Comment by kevinkit** [comment]
None of the issues above helped me, so I simply changed the line 58 to 63 to the following:

`mhd_flags = MHD_USE_SELECT_INTERNALLY`

**Comment by stefanofiorentino** [comment]
as @kevinkit said.. still in error for older ubuntu (14.04)
https://gist.github.com/stefanofiorentino/594858159c23841ec4eec56184852d51

**Comment by sharwell** [comment]
@cinemast We're seeing the same thing reported by @stefanofiorentino.

> libjson-rpc-cpp-1.1.0/src/jsonrpccpp/server/connectors/httpserver.cpp:47:35: error: ‘MHD_FEATURE_EPOLL’ was not declared in this scope

**Comment by stefanofiorentino** [comment]
@sharwell my posted [gist](https://gist.github.com/stefanofiorentino/594858159c23841ec4eec56184852d51) solved on Ubuntu 14.04 to me

**Comment by sharwell** [comment]
@stefanofiorentino I ended up recommending this approach:

```
install:
  - wget -c http://ftp.gnu.org/gnu/libmicrohttpd/libmicrohttpd-0.9.59.tar.gz
  - (tar zxf libmicrohttpd-0.9.59.tar.gz; cd libmicrohttpd-0.9.59; ./configure; make; sudo make install; cd ..; rm -fr libmicrohttpd-0.9.59)
  - wget -O libjson-rpc-cpp-1.1.0.tar.gz https://github.com/cinemast/libjson-rpc-cpp/archive/v1.1.0.tar.gz
  - (tar xzf libjson-rpc-cpp-1.1.0.tar.gz; cd libjson-rpc-cpp-1.1.0; cmake . && make; sudo make install; sudo ldconfig; cd ..; rm -fr libjson-rpc-cpp-1.1.0)
```

It was based on the following initial observation:

> It looks like the issue occurs if you don't have libmicrohttpd 0.9.39 or newer, per the following release notes:
>
> https://github.com/svn2github/libmicrohttpd/blob/e4a96ac6b99bc120b34c6af4ed9520d8b50c355e/ChangeLog#L522-L531
>
> It looks like 0.9.33-1 is installed on these machines by default. An approach like this, if it works, would be preferable:
>
> https://github.com/dupgit/sauvegarde/blob/fa5c0156aee008ec4ecd6b21a9c05d150652e7b8/.travis.yml#L28-L30
